marray/__init__.py

```diff
@@ -252,6 +253,17 @@ def fun(*args, name=name, **kwargs):
             return MaskedArray(out, mask=xp.any(masks, axis=0))
         setattr(mod, name, fun)
 
+
+    def clip(x, /, min=None, max=None):
+        args = [x, min, max]
+        masks = [arg.mask for arg in args if hasattr(arg, 'mask')]
+        masks = xp.broadcast_arrays(*masks)
+        mask = xp.any(masks, axis=0)
+        datas = [getattr(arg, 'data', arg) for arg in args]
+        data = xp.clip(datas[0], min=datas[1], max=datas[2])
+        return MaskedArray(data, mask)
+    mod.clip = clip
```

Comment:
this should be done in a new PR, but I believe it would be neater to use a decorator for this

Reply:
You mean for adding the attribute to the namespace? What is the advantage?